services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerEndpoint.java

```diff
@@ -392,11 +392,21 @@ private Object getCredential(UserSessionModel userSessionModel, String vcType, F
                 .filter(Objects::nonNull)
                 .toList();
 
-        VerifiableCredential credentialToSign = getVCToSign(protocolMappers, vcType, userSessionModel);
+        VCIssuanceContext vcIssuanceContext = getVCToSign(protocolMappers, credentialConfig, authResult, credentialRequestVO);
+
+        String specificConfigKey = VerifiableCredentialsSigningService.locator(credentialConfig.getFormat(), credentialConfig.getId());
+        String formatKey = credentialConfig.getFormat().name();
 
-        return Optional.ofNullable(signingServices.get(format))
-                .map(verifiableCredentialsSigningService -> verifiableCredentialsSigningService.signCredential(credentialToSign))
-                .orElseThrow(() -> new IllegalArgumentException(String.format("Requested format %s is not supported.", format)));
+        // First retrieve by format/id, then by format only
+        VerifiableCredentialsSigningService signingService = signingServices.getOrDefault(
+                specificConfigKey, signingServices.get(formatKey)
+        );
+
+        return Optional.ofNullable(signingService)
+                .map(service -> service.signCredential(vcIssuanceContext))
+                .orElseThrow(() -> new IllegalArgumentException(
```

Comment:
If this exception is thrown, which kind of an error response a holder receive?

Reply:
Resolved with [346b41](https://github.com/keycloak/keycloak/commit/346b4151539737e83126f8cc9705b462441c1081)